routing/index.ts

```diff
@@ -0,0 +1,183 @@
+import {createAction} from '../actions';
+import {withTask} from '../tasks';
+import {HISTORY_PUSH_TASK} from '../history';
+
+type ComponentType = any;
+
+type RouteDefinition = {
+  [route: string]: ComponentType | [ComponentType, RouteDefinition]
+}
+
+/*
+ * Example usage:
+ *
+ * createRouter({
+ *   '/': Index,
+ *   '/users': [User, {
+ *     '/'
+ *   }]
+ * })
+ */
+export function createRouter(routes: RouteDefinition, onChange = (state) => state) {
+  const LOCATION_CHANGE = createAction('LOCATION_CHANGE');
```

Comment:
These are intentionally created here; you might want to have multiple "routers" in your app, for instance.